processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -159,8 +159,18 @@ public GroupByQueryQueryToolChest(
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
       final List<AggregatorFactory> aggs = Lists.newArrayList();
+
       for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
```

Comment:
@dcim for the query, did you have broker query going to multiple historicals and some merging happening at the broker? I believe in that case the result produced with this PR will be wrong for one of the aggregators as IncrementalIndex would use one of them to do the aggregation and will not produce correct results for the other one.